File: 2024/03A.py
import sys
import re

def solve(input):
    return sum([int(a) * int(b) for (a,b) in re.findall(r'mul\((\d+),(\d+)\)', input)])      
        
if __name__ == '__main__':
    print(solve(open(sys.argv[1]).read())) 

File: 2024/03A.js
function solve(input) {
  let result = 0;
  for (match of input.matchAll(/mul\((\d+),(\d+)\)/g)) {
    result += match[1] * match[2];
  }
  return result;
}

console.log(solve(require('node:fs').readFileSync(process.argv[2], 'utf8').trim()));